Build of AppVeyor image for Windows target fails

Hydrogen branch          : `development`
Operating system + version  : Windows

- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -

Since a couple of weeks the AppVeyor image of the Windows build does constantly fail with the following error message:

> C:\projects\hydrogen\src\core\AudioEngine.cpp:23:10: error: 'core/AudioEngine.h' file not found [clang-diagnostic-error]
> #include <core/AudioEngine.h>
>         ^
> 1 error generated.

e.g. in https://ci.appveyor.com/project/mauser/hydrogen/builds/39176767/job/r3pciwt1o7iw9iix

I took a look into the `.appveyor.yml` file - without understanding the peculiarities of the framework - but couldn't find anything suspicious?

@mauser could you have a look at it? Seems to be introduced by the refactoring of the Audio Engine.

--- a/src/core/Basics/Song.cpp
+++ b/src/core/Basics/Song.cpp
@@ -745,7 +745,7 @@ QString Song::toQString( const QString& sPrefix, bool bShort ) const {
 			.append( QString( ", m_sPlaybackTrackFilename: %1" ).arg( m_sPlaybackTrackFilename ) )
 			.append( QString( ", m_bPlaybackTrackEnabled: %1" ).arg( m_bPlaybackTrackEnabled ) )
 			.append( QString( ", m_fPlaybackTrackVolume: %1" ).arg( m_fPlaybackTrackVolume ) )
-			.append( QString( "%1" ).arg( m_pVelocityAutomationPath->toQString( sPrefix + s ) ) )
+			.append( QString( ", m_pVelocityAutomationPath: %1" ).arg( m_pVelocityAutomationPath->toQString( sPrefix ) ) )
 			.append( QString( ", m_sLicense: %1" ).arg( m_sLicense ) );
 		if ( m_actionMode == ActionMode::selectMode ) {
 			sOutput.append( QString( ", m_actionMode: 0" ) );

--- a/src/core/Hydrogen.cpp
+++ b/src/core/Hydrogen.cpp
@@ -35,12 +35,10 @@
 #include <deque>
 #include <queue>
 #include <iostream>
-#include <ctime>
 #include <cmath>
 #include <algorithm>
 #include <thread>
 #include <chrono>
-#include <ctime>
 
 #include <QtCore/QMutex>
 #include <QtCore/QMutexLocker>
@@ -1773,7 +1771,7 @@ QString Hydrogen::toQString( const QString& sPrefix, bool bShort ) const {
 		for ( auto dd : m_nBeatDiffs ) {
 			sOutput.append( QString( " %1" ).arg( dd ) );
 		}
-		sOutput.append( QString( "]\n%1%2m_CurrentTime: %3" ).arg( sPrefix ).arg( s ).arg( std::ctime( &m_CurrentTime.tv_sec ) ) )
+		sOutput.append( QString( "]\n%1%2m_CurrentTime: %3" ).arg( sPrefix ).arg( s ).arg( static_cast<long>(m_CurrentTime.tv_sec ) ) )
 			.append( QString( "%1%2m_nCoutOffset: %3\n" ).arg( sPrefix ).arg( s ).arg( m_nCoutOffset ) )
 			.append( QString( "%1%2m_nStartOffset: %3\n" ).arg( sPrefix ).arg( s ).arg( m_nStartOffset ) )
 			.append( QString( "%1%2m_oldEngineMode: %3\n" ).arg( sPrefix ).arg( s ).arg( m_oldEngineMode ) )
@@ -1798,14 +1796,12 @@ QString Hydrogen::toQString( const QString& sPrefix, bool bShort ) const {
 		}
 		sOutput.append( QString( "%1%2m_fNewBpmJTM: %3\n" ).arg( sPrefix ).arg( s ).arg( m_fNewBpmJTM ) )
 			.append( QString( "%1%2m_nSelectedInstrumentNumber: %3\n" ).arg( sPrefix ).arg( s ).arg( m_nSelectedInstrumentNumber ) )
-			.append( QString( "%1%2m_pAudioEngine: \n" ) )//.arg( sPrefix ).arg( s ).arg( m_pAudioEngine ) )
+			.append( QString( "%1%2m_pAudioEngine: \n" ).arg( sPrefix ).arg( s ) )//.arg( m_pAudioEngine ) )
 			.append( QString( "%1%2lastMidiEvent: %3\n" ).arg( sPrefix ).arg( s ).arg( lastMidiEvent ) )
 			.append( QString( "%1%2lastMidiEventParameter: %3\n" ).arg( sPrefix ).arg( s ).arg( lastMidiEventParameter ) )
-			.append( QString( "%1%2m_nInstrumentLookupTable: [" ).arg( sPrefix ).arg( s ) );
-		for ( auto ll : m_nInstrumentLookupTable ) {
-			sOutput.append( QString( " %1" ).arg( ll ) );
-		}
-		sOutput.append( QString( "]\n%1%2m_nMaxTimeHumanize: %3\n" ).arg( sPrefix ).arg( s ).arg( m_nMaxTimeHumanize ) );
+			.append( QString( "%1%2m_nInstrumentLookupTable: [ %3 ... %4 ]\n" ).arg( sPrefix ).arg( s )
+					 .arg( m_nInstrumentLookupTable[ 0 ] ).arg( m_nInstrumentLookupTable[ MAX_INSTRUMENTS ] ) )
+			.append( QString( "%1%2m_nMaxTimeHumanize: %3\n" ).arg( sPrefix ).arg( s ).arg( m_nMaxTimeHumanize ) );
 	} else {
 		
 		sOutput = QString( "%1[Hydrogen]" ).arg( sPrefix )
@@ -1824,7 +1820,7 @@ QString Hydrogen::toQString( const QString& sPrefix, bool bShort ) const {
 		for ( auto dd : m_nBeatDiffs ) {
 			sOutput.append( QString( " %1" ).arg( dd ) );
 		}
-		sOutput.append( QString( "], m_CurrentTime: %1" ).arg( std::ctime( &m_CurrentTime.tv_sec ) ) )
+		sOutput.append( QString( "], m_CurrentTime: %1" ).arg( static_cast<long>( m_CurrentTime.tv_sec ) ) )
 			.append( QString( ", m_nCoutOffset: %1" ).arg( m_nCoutOffset ) )
 			.append( QString( ", m_nStartOffset: %1" ).arg( m_nStartOffset ) )
 			.append( QString( ", m_oldEngineMode: %1" ).arg( m_oldEngineMode ) )
@@ -1833,7 +1829,7 @@ QString Hydrogen::toQString( const QString& sPrefix, bool bShort ) const {
 			.append( QString( ", m_GUIState: %1" ).arg( static_cast<int>( m_GUIState ) ) );
 		sOutput.append( QString( ", m_pTimeline: " ) );
 		if ( m_pTimeline != nullptr ) {
-			sOutput.append( QString( "%1" ).arg( m_pTimeline->toQString( sPrefix + s, bShort ) ) );
+			sOutput.append( QString( "%1" ).arg( m_pTimeline->toQString( sPrefix, bShort ) ) );
 		} else {
 			sOutput.append( QString( "nullptr" ) );
 		}						 
@@ -1852,11 +1848,9 @@ QString Hydrogen::toQString( const QString& sPrefix, bool bShort ) const {
 			.append( QString( ", m_pAudioEngine: " ) )// .arg( m_pAudioEngine ) )
 			.append( QString( ", lastMidiEvent: %1" ).arg( lastMidiEvent ) )
 			.append( QString( ", lastMidiEventParameter: %1" ).arg( lastMidiEventParameter ) )
-			.append( QString( ", m_nInstrumentLookupTable: [" ) );
-		for ( auto ll : m_nInstrumentLookupTable ) {
-			sOutput.append( QString( " %1" ).arg( ll ) );
-		}
-		sOutput.append( QString( "], m_nMaxTimeHumanize: %1\n" ).arg( m_nMaxTimeHumanize ) );
+			.append( QString( ", m_nInstrumentLookupTable: [ %1 ... %2 ]" )
+					 .arg( m_nInstrumentLookupTable[ 0 ] ).arg( m_nInstrumentLookupTable[ MAX_INSTRUMENTS ] ) )
+			.append( QString( ", m_nMaxTimeHumanize: %1\n" ).arg( m_nMaxTimeHumanize ) );
 	}
 		
 	return sOutput;

--- a/src/core/Timeline.cpp
+++ b/src/core/Timeline.cpp
@@ -168,13 +168,13 @@ QString Timeline::toQString( const QString& sPrefix, bool bShort ) const {
 	} else {
 		
 		sOutput = QString( "%1[Timeline] " ).arg( sPrefix )
-			.append( QString( "m_tempoMarkers: [" ).arg( sPrefix ).arg( s ) );
+			.append( QString( "m_tempoMarkers: [" ) );
 		for ( auto const& tt : m_tempoMarkers ) {
 			if ( tt != nullptr ) {
 				sOutput.append( QString( " [bar: %1 , bpm: %2]" ).arg( tt->nBar ).arg( tt->fBpm ) );
 			}
 		}
-		sOutput.append( QString( "], m_tags: [" ).arg( sPrefix ).arg( s ) );
+		sOutput.append( QString( "], m_tags: [" ) );
 		for ( auto const& tt : m_tags ) {
 			if ( tt != nullptr ) {
 				sOutput.append( QString( " [bar: %1 , tag: %2]" ).arg( tt->nBar ).arg( tt->sTag ) );

